Feature Request: Support `InlineArray` type sugar

## Summary

Swift can use experimental type sugar for `InlineArray` (e.g. `[3 x Int]`), and swift-syntax already has supported it, but swift-format hasn't supported it yet.

- [pitch](https://forums.swift.org/t/pitch-inlinearray-type-sugar/79142)
- swiftlang/swift: https://github.com/swiftlang/swift/pull/80087
- swiftlang/swift-syntax: https://github.com/swiftlang/swift-syntax/pull/3021

## Input

`test.swift`:

```swift
let a: [3 x Int] = [1, 2, 3]
```

## Expected result

```swift
let a: [3 x Int] = [1, 2, 3]
```

## Actual result

(with `--enable-experimental-feature InlineArrayTypeSugar`)

```swift
let a: [3xInt] = [1, 2, 3]
```

However, this result causes compiler errrors.

```console
$ swift-format format --enable-experimental-feature InlineArrayTypeSugar -i test.swift
$ swiftc -enable-experimental-feature InlineArrayTypeSugar test.swift
test.swift:1:10: error: 'x' is not a valid digit in integer literal
1 | let a: [3xInt] = [1, 2, 3]
  |          `- error: 'x' is not a valid digit in integer literal
2 | 

test.swift:1:9: error: expected element type
1 | let a: [3xInt] = [1, 2, 3]
  |         `- error: expected element type
2 | 

test.swift:1:9: error: expected ']' in array type
1 | let a: [3xInt] = [1, 2, 3]
  |        |`- error: expected ']' in array type
  |        `- note: to match this opening '['
2 | 

test.swift:1:14: error: consecutive statements on a line must be separated by ';'
1 | let a: [3xInt] = [1, 2, 3]
  |              `- error: consecutive statements on a line must be separated by ';'
2 | 

test.swift:1:14: error: expected expression
1 | let a: [3xInt] = [1, 2, 3]
  |              `- error: expected expression
2 | 
```

## Environment

- swift-format: 58807b9061310e626e8be9a2831be2fb7ef99858
- swift-syntax: [820501e3cf5894ad1deb8de95649b938a8bbe9f4](https://github.com/swiftlang/swift-syntax/commit/820501e3cf5894ad1deb8de95649b938a8bbe9f4)

--- a/Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift
+++ b/Sources/SwiftFormat/PrettyPrint/TokenStreamCreator.swift
@@ -1732,6 +1732,14 @@ fileprivate final class TokenStreamCreator: SyntaxVisitor {
     return .visitChildren
   }
 
+  override func visit(_ node: InlineArrayTypeSyntax) -> SyntaxVisitorContinueKind {
+    after(node.leftSquare, tokens: .break(.open, size: 0), .open)
+    before(node.separator, tokens: .space)
+    after(node.separator, tokens: .break)
+    before(node.rightSquare, tokens: .break(.close, size: 0), .close)
+    return .visitChildren
+  }
+
   override func visit(_ node: TupleTypeSyntax) -> SyntaxVisitorContinueKind {
     after(node.leftParen, tokens: .break(.open, size: 0), .open)
     before(node.rightParen, tokens: .break(.close, size: 0), .close)

--- a/Tests/SwiftFormatTests/PrettyPrint/ArrayDeclTests.swift
+++ b/Tests/SwiftFormatTests/PrettyPrint/ArrayDeclTests.swift
@@ -302,4 +302,72 @@ final class ArrayDeclTests: PrettyPrintTestCase {
 
     assertPrettyPrintEqual(input: input, expected: expected, linelength: 32)
   }
+
+  func testInlineArrayTypeSugar() {
+    let input =
+      """
+      let a: [3 of Int]
+      let a: [[3 of Int]]
+      let a: [3 of [3 of Int]]
+      let a: [n of Int]
+      let fiveIntegers: [5 of _] = .init(repeating: 99)
+      let fourBytes: [_ of Int8] = [1, 2, 3, 4]
+      let fourIntegers: [_ of _] = [1, 2, 3, 4]
+      let fiveDoubles = [5 of _](repeating: 1.23)
+
+      """
+
+    let expected =
+      """
+      let a: [3 of Int]
+      let a: [[3 of Int]]
+      let a: [3 of [3 of Int]]
+      let a: [n of Int]
+      let fiveIntegers: [5 of _] = .init(repeating: 99)
+      let fourBytes: [_ of Int8] = [1, 2, 3, 4]
+      let fourIntegers: [_ of _] = [1, 2, 3, 4]
+      let fiveDoubles = [5 of _](repeating: 1.23)
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 80)
+  }
+
+  func testInlineArrayTypeSugarWhenLineLengthExceeded() {
+    let input =
+      """
+      let a: [3 of VeryLongGenericTypeNameThatCausesWrapping]
+      let a: [3 of [3 of VeryLongGenericTypeNameThatCausesWrapping]]
+      let a = [3 of VeryLongGenericTypeNameThatCausesWrapping](repeating: foo)
+
+      """
+
+    let expected =
+      """
+      let a:
+        [
+          3 of
+            VeryLongGenericTypeNameThatCausesWrapping
+        ]
+      let a:
+        [
+          3 of
+            [
+              3 of
+                VeryLongGenericTypeNameThatCausesWrapping
+            ]
+        ]
+      let a =
+        [
+          3 of
+            VeryLongGenericTypeNameThatCausesWrapping
+        ](
+          repeating:
+            foo
+        )
+
+      """
+
+    assertPrettyPrintEqual(input: input, expected: expected, linelength: 5)
+  }
 }

